pandas/core/internals.py

```diff
@@ -206,7 +206,7 @@ def array_dtype(self):
         """
         return self.dtype
 
-    def make_block(self, values, placement=None, ndim=None):
+    def make_block(self, values, placement=None, ndim=None, dtype=None):
```

Comment:
you can add this back, but it needs to be deprecated anyhow.

Reply:
Then please answer in the original issue how to achieve the same result (see the code example I posted there, to create a new DatetimeTZ block from a numpy array). 
But I don't see why this needs to be deprecated. The `dtype` argument was actually doing something, in contrast to `fastpath` (which we therefore deprecated)

Reply:
fp should be using ``make_block`` and not ``make_block_same_class``